components/member_list_channel/member_list_channel.jsx

```diff
@@ -144,14 +145,18 @@ export default class MemberListChannel extends React.Component {
                 const user = users[i];
 
                 if (teamMembers[user.id] && channelMembers[user.id] && user.delete_at === 0) {
-                    usersToDisplay.push(user);
+                    const status = UserStore.getStatus(user.id);
+                    usersToDisplay.push({...user, status});
+
                     actionUserProps[user.id] = {
                         channel: this.props.channel,
                         teamMember: teamMembers[user.id],
                         channelMember: channelMembers[user.id]
                     };
                 }
             }
+
+            usersToDisplay.sort(Utils.sortUsersByStatusAndDisplayName);
```

Comment:
We should probably look into moving a bunch of this logic out of the render function if possible. It's possible that we'll render multiple times without the usersToDisplay changing, so rebuilding and sorting, and setting the result to state (or props even) might be worthwhile here. I'm only 1/5 on this though, it's going a bit beyond the scope of the ticket but if a user list is 1000+ in length this could potentially help performance a lot

Turning it into a selector would be even better (though more work)